### DISCUSSION TITLE
Hybrid-solver: Setting a spatial density profile for electrons

### DISCUSSION THREAD

**Discussion by cooluser26** [BLAST-WarpX/warpx#4649] [Q&A]
Hello everyone,

I'm looking to model ion-beam propagation in a plasma, where the electron density decays exponentially with distance. Is it possible to set the density profile for the electron? I saw that there is an input term called `hybrid_pic_model.n0_ref (float) ` but does this refer to the electron density?  Would creating another species called "electrons" for my purpose?

Any help would be greatly appreciated, thank you!!

**Comment by roelof-groenewald** [comment]
Hi,

The hybrid-PIC model has a built in assumption of quasi-neutrality so with that solver the electron density will always be assumed equal to the sum of all ion densities. 

If you just want to have a static electron background population you can load electrons with the desired density distribution but add the flag `do_not_push` to the electron species initialization. In this setup the electrons will not be pushed in time and therefore their density distribution will not change. If you are using the electrostatic solver, the electron density will still be accounted for during the Poisson solve though.

Hope that helps.
